lib/active_force/sobject.rb

```diff
@@ -222,9 +222,18 @@ def logger_output action, exception, params = {}
     end
 
     def attributes_for_sfdb
+      persisted? ? sfdb_attributes_for_update : sfdb_attributes_for_create
+    end
+
+    def sfdb_attributes_for_create
+      attrs_to_change = changed
+      @attributes.each_value {|value| attrs_to_change << value.name if value.class == ActiveModel::Attribute::UserProvidedDefault }
```

Comment:
This is a good callout and something that operates slightly differently than I expected as well. Here is some more context:

```
# File activemodel/lib/active_model/dirty.rb, line 173
def changed
  mutations_from_database.changed_attribute_names
end
```

In the AF code snipped, I set the return value of `changed` to a variable so I can add to it.

ie.

![Screenshot 2023-05-25 at 8 40 13 AM](https://github.com/Beyond-Finance/active_force/assets/87046098/f43c79a9-accd-4e40-8290-26d2cf2b0724)

Reply:
Oh interesting.  It re-evaluates on each call.  Might still be clearer anyway to use non-mutating array methods instead of appending to an array as you loop over `@attributes`.  I was thinking something like (not sure if you need the `#to_a`):

```ruby
attr_to_change = @attributes.each_value.to_a.select { |value| value.is_a?(ActiveModel::Attribute::UserProvidedDefault) }
                                            .map(&:name)
                                            .concat(changed)
```

Reply:
That seems to avoid the variable set nicely! (without .to_a). I'm not as familiar with chaining enumerables like this, are there any warnings that go along with it? Ie. performance?

Reply:
You are iterating over the enumerable for each method. You'll touch each item 3 times in the worst case if nothing is filtered by `select` (although, not sure how exactly concat works, so it might be less than that).  If you combined the whole thing in one loop like you had before, you'd have fewer operations.  But, we're talking about ~100 items in the worst case here?  I doubt it makes any noticeable difference even up to 10K items.